unit-tests/live/frames/test-fps-permutations.py

```diff
@@ -0,0 +1,330 @@
+        sensor.open(profiles)
+        sensor.start(funcs_dict[sensor])
+
+    # the core of the test - frames are counted during sleep when count_frames is on
+    time.sleep(TIME_FOR_STEADY_STATE)
+    count_frames = True  # Start counting frames
+    time.sleep(TIME_TO_COUNT_FRAMES)
+    count_frames = False  # Stop counting
+
+    for sensor, profiles in sensor_profiles_dict.items():
+        for profile in profiles:
+            profile_name_fps_dict[profile.stream_name()] /= TIME_TO_COUNT_FRAMES
+
+        sensor.stop()
+        sensor.close()
+
+    return profile_name_fps_dict
+
+
+def get_test_details_str(sensor_profile_dict):
+    s = ""
+    for sensor, profiles in sensor_profile_dict.items():
+        for profile in profiles:
+            s += (f"Expected fps for profile {profile.stream_name()} on sensor "
+                  f"{sensor.name} is {profile.fps()} "
+                  f"on {get_resolution(profile)}\n")
+
+    s = s.replace("on (0, 0)", "")  # remove no resolution for Motion Module profiles
+    return s
+
+
+def perform_fps_test(sensor_profiles_arr, modes):
+
+    log.d(get_time_est_string(len(sensor_profiles_arr), modes))
+
+    for mode in modes:
+        # Generate a list of all possible combinations of 1s and 0s (permutations) in the length of the array
+        perms = list(itertools.product([0, 1], repeat=len(sensor_profiles_arr)))
+
+        # Go over every possible permutation and check if we should test it
+        # Each index in a permutation represents a profile, which will be tested only if the value in that index is 1
+        for perm in perms:
+            # log.d(perm)
+            if all(v == 0 for v in perm):
+                continue
+            if should_test(mode, perm):
+                partial_dict = get_dict_for_permutation(sensor_profiles_arr, perm)
+                test.start("Testing", get_tested_profiles_string(partial_dict))
+                expected_fps = get_expected_fps(partial_dict)
+                log.d(get_test_details_str(partial_dict))
+                fps_dict = measure_fps(partial_dict)
+                test.check(check_fps_dict(fps_dict, expected_fps))
+                test.finish()
+
+
+####################################################
+# ---------- Test Preparation Functions ---------- #
+####################################################
+def get_profiles_by_resolution(sensor, resolution, fps=None):
+    # resolution is a required parameter because on a sensor all active profiles must have the same resolution
+    profiles = []
+    stream_types_added = []
+    for p in sensor.get_stream_profiles():
+        if get_resolution(p) == resolution:
+            if fps is None or p.fps() == fps:
+                # to avoid having a long run time, we don't choose the same stream more than once
+                if p.stream_type() not in stream_types_added:
+                    if p.stream_type() == rs.stream.infrared:
+                        if p.stream_index() != 1:
```

Comment:
Style comment (decide if you want to change) - this is very indented you can use `and` instead of the last `if`, or better yet, move the last 3 `if`s to a function `should_add_profile`